train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -43,7 +47,16 @@ private void enforceSpeedLimit() {
 
 	@Override
 	public void setJoystickPosition(int joystickPosition) {
-		this.step = joystickPosition;		
+		this.step = joystickPosition;
+		if(timer != null)
+			timer.cancel();
+		timer = new Timer();
+		timer.schedule(new TimerTask() {
```

Comment:
You might want to consider extracting your lamda function into an existing private function, just saying....

Reply:
Nah it's okay for now